includes/class-wc-stripe-payment-method-configurations.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class WC_Stripe_Payment_Method_Configurations
+ */
+class WC_Stripe_Payment_Method_Configurations {
+	/**
+	 * Get the merchant payment method configuration in Stripe.
+	 *
+	 * @return object|null
+	*/
+	private static function get_primary_configuration() {
+		$result = WC_Stripe_API::get_instance()->get_payment_method_configurations();
+		$payment_method_configurations = $result->data ?? null;
+
+		if ( ! $payment_method_configurations ) {
+			return null;
+		}
+
+		foreach ( $payment_method_configurations as $payment_method_configuration ) {
+			if ( $payment_method_configuration->active && $payment_method_configuration->parent ) {
+				return $payment_method_configuration;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	* Get the UPE enabled payment method IDs.
+	*
+	* @return array
+	*/
+	public static function get_upe_enabled_payment_method_ids() {
+		$upe_enabled_payment_method_ids = get_transient( 'woocommerce_stripe_upe_enabled_payment_method_ids' );
+
+		if ( $upe_enabled_payment_method_ids ) {
+			return $upe_enabled_payment_method_ids;
+		}
+
+		$enabled_payment_method_ids            = [];
+		$merchant_payment_method_configuration = self::get_primary_configuration();
+
+		if ( $merchant_payment_method_configuration ) {
+			foreach ( $merchant_payment_method_configuration as $payment_method_id => $payment_method ) {
+				if ( is_object( $payment_method ) &&
+					property_exists( $payment_method, 'display_preference' ) &&
+					property_exists( $payment_method->display_preference, 'value' ) ) {
+
+					$payment_method_status = 'on' === $payment_method->display_preference->value;
+
+					if ( $payment_method_status ) {
+						$enabled_payment_method_ids[] = $payment_method_id;
+					}
+				}
+			}
+		}
+
+		set_transient( 'woocommerce_stripe_upe_enabled_payment_method_ids', $enabled_payment_method_ids, DAY_IN_SECONDS );
```

Comment:
We cache the [account response data for 2 hours](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/trunk/includes/class-wc-stripe-account.php#L132). We can use the same expiration time here.

Reply:
On second thought, 2 hours also seems a bit long. When I enable any method on Stripe dashboard, it does not update in wp-admin because of the caching.

Reply:
I removed the transient in this PR to meet the expectation. We can change it later if needed.